hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/clean/CleanPlanner.java

```diff
@@ -146,11 +146,15 @@ public CleanPlanner(HoodieEngineContext context, HoodieTable<T, I, K, O> hoodieT
     if (config.incrementalCleanerModeEnabled()) {
       Option<HoodieInstant> lastClean = hoodieTable.getCleanTimeline().filterCompletedInstants().lastInstant();
       if (lastClean.isPresent()) {
-        HoodieCleanMetadata cleanMetadata = TimelineMetadataUtils
-            .deserializeHoodieCleanMetadata(hoodieTable.getActiveTimeline().getInstantDetails(lastClean.get()).get());
-        if ((cleanMetadata.getEarliestCommitToRetain() != null)
-            && (cleanMetadata.getEarliestCommitToRetain().length() > 0)) {
-          return getPartitionPathsForIncrementalCleaning(cleanMetadata, instantToRetain);
+        if (hoodieTable.getActiveTimeline().isEmpty(lastClean.get())) {
```

Comment:
again, can you help me understand when this could happen ?

Reply:
@nsivabalan Hi,This merged [PR](https://github.com/apache/hudi/pull/1128) is very similar to it. Maybe we can find the reason from it

Reply:
sure, will take a look. looks like for some of the actions, requested meta file is empty and hence. but I would expect the deserialize should have known the specifics of which one can be empty and which ones cannot. need to understand bit more on this.

Reply:
Note to self: so here we just delete the completed instant leaving requested and inflight as is in timeline.